File: src/main/java/io/lcydev/spawnercleaner/RemoveTask.java
package io.lcydev.spawnercleaner;

import io.lcydev.spawnercleaner.utils.myLogger;
import org.bukkit.Location;
import org.bukkit.Material;

public class RemoveTask implements Runnable {
    private static SpawnerCleaner instance;
    private static myLogger logger;

    public RemoveTask() {
        instance = SpawnerCleaner.inst();
        logger = myLogger.getLog();
    }

    @Override
    public void run() {
        Location spawnerLoc = instance.removeQueue.poll();
        if (spawnerLoc != null) {
            spawnerLoc.getBlock().setType(Material.AIR);
            logger.fine(String.format("Removed spawner at %s: %.1f %.1f %.1f", spawnerLoc.getWorld(), spawnerLoc.getX(), spawnerLoc.getY(), spawnerLoc.getZ()));
        }
    }
}

File: src/main/kotlin/io/lcydev/spawnercleaner/tasks/RemoveTask.kt
package tasks

import io.lcydev.template.SpawnerCleaner
import org.bukkit.Location
import org.bukkit.Material

class RemoveTask(private val plugin: SpawnerCleaner) : Runnable {

    override fun run() {
        val location: Location = plugin.removeQueue.poll() ?: return

        location.block.type = Material.AIR
        plugin.slF4JLogger.info("Removed ${location.block.type} in world ${location.world} at ${location.blockX}, ${location.blockY}, ${location.blockZ}")
    }
}